edward/util.py

```diff
@@ -133,81 +168,152 @@ def kl_multivariate_normal(loc_one, scale_one, loc_two=0, scale_two=1):
+        lengthscale of radial basis function
+
+    Returns
+    -------
+    tf.Tensor
+        scalar if vector input, rank-(n-1) if n-Tensor input
     """
     return tf.pow(sigma, 2.0) * \
            tf.exp(-1.0/(2.0*tf.pow(l, 2.0)) * \
                   tf.reduce_sum(tf.pow(x - y , 2.0)))
 
 def rbf(x, y=0.0, sigma=1.0, l=1.0):
-    """
-    Squared-exponential kernel element-wise
-    k(x, y) = sigma^2 exp{ -1/(2l^2) (x_i - y_i)^2 }
+    """Squared-exponential kernel element-wise
+
+    .. math:: k(x, y) = \sigma^2 \exp{ -1/(2l^2) (x_i - y_i)^2 }
+
+    Parameters
+    ----------
+    x : tf.Tensor
+        scalar, vector, matrix, or n-Tensor 
+    y : Optional[tf.Tensor], default 0.0
+        scalar, vector, matrix, or n-Tensor 
+    sigma : Optional[double], default 1.0
+        standard deviation of radial basis function
+    l : Optional[double], default 1.0
+        lengthscale of radial basis function
+
+    Returns
+    -------
+    tf.Tensor
+        size corresponding to size of input
     """
     return tf.pow(sigma, 2.0) * \
            tf.exp(-1.0/(2.0*tf.pow(l, 2.0)) * tf.pow(x - y , 2.0))
 
 def set_seed(x):
-    """
-    Set seed for both NumPy and TensorFlow.
+    """Set seed for both NumPy and TensorFlow.
+
+    Parameters
+    ----------
+    x : double
+        seed
     """
     np.random.seed(x)
     tf.set_random_seed(x)
 
 def softplus(x):
-    """
-    Softplus. TensorFlow can't currently autodiff through
-    tf.nn.softplus().
+    """Elementwise Softplus function
+
+    .. math:: \log(1 + \exp(x))
+
+    TensorFlow can't currently autodiff through ``tf.nn.softplus()``.
+
+    Parameters
+    ----------
+    x : tf.Tensor
+        scalar, vector, matrix, or n-Tensor 
+
+    Returns
+    -------
+    tf.Tensor
+        size corresponding to size of input
     """
     return tf.log(1.0 + tf.exp(x))
 
 def stop_gradient(x):
-    """
-    Apply tf.stop_gradient() element-wise.
+    """Apply ``tf.stop_gradient()`` element-wise.
+
+    Parameters
+    ----------
+    x : tf.Tensor or list
+        scalar, vector, matrix, or n-Tensor or list thereof
+        
+    Returns
+    -------
+    tf.Tensor or list
+        size corresponding to size of input
+
     """
     if isinstance(x, tf.Tensor):
         return tf.stop_gradient(x)
     else: # list
         return [tf.stop_gradient(i) for i in x]
 
 def to_simplex(x):
-    """
-    Transform real vector of length (K-1) to a simplex of dimension K
+    """Transform real vector of length ``(K-1)`` to a simplex of dimension ``K``
     using a backward stick breaking construction.
 
     Parameters
     ----------
     x : tf.tensor or np.array
-        Vector or matrix.
+        scalar, vector, or matrix
 
     Returns
     -------
     tf.Tensor
-        Same shape as input but with last dimension of size K.
-
-    Notes
-    -----
-    x as a 3d or higher tensor is not guaranteed to be supported.
+        Same shape as input but with last dimension of size ``K``.
```

Comment:
Can you keep that Notes section about not guaranteeing support? I find it personally useful for knowing what needs to be extended. Or perhaps to keep consistency those Notes should be added as github issues